stake-pool/js/src/index.ts

```diff
@@ -0,0 +1,104 @@
+import * as schema from './schema.js';
+import borsh from 'borsh';
+import solanaWeb3 from '@solana/web3.js';
+
+export class StakePool {
+  /**
+   * Wrapper class for a stake pool.
+   * Each stake pool has a stake pool account and a validator list account.
+   * (Optionally) a stake pool can also have a name and ticker.
+   */
+  name: string;
+  ticker: string;
+  stakePool: {
+    pubkey: solanaWeb3.PublicKey;
+    accountInfo: solanaWeb3.AccountInfo<schema.StakePoolAccount>;
+  };
+  validatorList: {
+    pubkey: solanaWeb3.PublicKey;
+    accountInfo: solanaWeb3.AccountInfo<schema.ValidatorListAccount>;
+  };
+}
+
+function decodeSerializedStakePool(
+  serializedStakePool: Buffer,
+  accountType,
+): schema.StakePoolAccount | schema.ValidatorListAccount {
+  return accountType.decode(serializedStakePool);
+}
+
+async function getStakePoolAccounts(
+  connection: solanaWeb3.Connection,
+  stakePoolAddress: solanaWeb3.PublicKey,
+): Promise<
+  {
+    account: solanaWeb3.AccountInfo<
+      schema.StakePoolAccount | schema.ValidatorListAccount
+    >;
+    pubkey: solanaWeb3.PublicKey;
+  }[]
+> {
+  try {
+    let response = await connection.getProgramAccounts(STAKE_POOL_ADDR);
+
+    const stakePoolAccounts = response.map(a => {
+      let b = {
+        pubkey: a.pubkey,
+        account: {
+          data: null,
+          executable: a.account.executable,
+          lamports: a.account.lamports,
+          owner: a.account.owner,
+        },
+      };
+
+      if (a.account.data.length === STAKE_POOL_ACCT_LENGTH) {
```

Comment:
We shouldn't have to do this, it defeats the whole purpose of a deserializing library like borsh!  I see two possible situations:
* we know the type being deserialized, so we just use it's decode right away
* we really don't know, in which case `AccountType` can actually have a value of `StakePool` / `ValidatorList`. You would just need to omit the account type enum at the start.  We shouldn't really run into that situation though.  If we need to do `getProgramAccounts`, we can use a filter on the first byte, either it equals 1 (StakePool), or 2 (ValidatorList)

Reply:
Fair enough --- love the idea to filter on the first byte. I've done that now.